crates/store/re_dataframe/tests/static.rs

```diff
@@ -0,0 +1,109 @@
+use std::sync::Arc;
+
+use arrow::array::{ListArray, RecordBatchIterator, StringArray, UInt32Array};
+use re_chunk::{ArrowArray as _, Chunk, RowId, TimePoint};
+use re_chunk_store::{ChunkStore, ChunkStoreConfig, QueryExpression};
+use re_dataframe::QueryEngine;
+use re_log_types::EntityPath;
+use re_types::AsComponents as _;
+use re_types::{AnyValues, ComponentDescriptor};
+
+#[test]
+fn query_static_columns() -> anyhow::Result<()> {
+    re_log::setup_logging();
+
+    let store = ChunkStore::new_handle(
+        re_log_types::StoreId::random(re_log_types::StoreKind::Recording),
+        ChunkStoreConfig::COMPACTION_DISABLED,
+    );
+
+    let any_values = AnyValues::default()
+        .with_field("yak", Arc::new(StringArray::from(vec!["yuk"])))
+        .with_field("foo", Arc::new(StringArray::from(vec!["bar"])))
+        .with_field("baz", Arc::new(UInt32Array::from(vec![42u32])));
+
+    let entity_path = EntityPath::from("test");
+
+    let chunk0 = Chunk::builder(entity_path.clone())
+        .with_serialized_batches(
+            RowId::new(),
+            TimePoint::default(),
+            any_values.as_serialized_batches(),
+        )
+        .build()?;
+
+    store.write().insert_chunk(&Arc::new(chunk0))?;
+
+    let engine = QueryEngine::from_store(store);
+
+    let query_expr = QueryExpression {
+        view_contents: None,
+        include_semantically_empty_columns: false,
+        include_indicator_columns: false,
+        include_tombstone_columns: false,
+        include_static_columns: re_chunk_store::StaticColumnSelection::Both,
+        filtered_index: None,
+        filtered_index_range: None,
+        filtered_index_values: None,
+        using_index_values: None,
+        filtered_is_not_null: None,
+        sparse_fill_strategy: re_chunk_store::SparseFillStrategy::None,
+        selection: None,
+    };
+
+    let query_handle = engine.query(query_expr);
+
+    let components = &query_handle.view_contents().components;
+
+    assert_eq!(
+        components
+            .iter()
+            .map(|column| (column.component_descriptor(), column.is_static))
+            .collect::<Vec<_>>(),
+        vec![
+            (ComponentDescriptor::partial("baz"), true),
+            (ComponentDescriptor::partial("foo"), true),
+            (ComponentDescriptor::partial("yak"), true),
+        ]
+    );
+
+    let schema = query_handle.schema().clone();
+
+    let mut reader = RecordBatchIterator::new(query_handle.into_batch_iter().map(Ok), schema);
+
+    let batch = reader.next().expect("there should be at least one batch")?;
```

Comment:
Using insta is even more advisable here, because of how annoying and partial the following boilerplate is. See e.g. `dataplatform/redap_manifest_registry/tests/` where it's used a lot.